--- tarsnaps.sh
+++ tarsnaps.sh
@@ -25,8 +25,9 @@
 backupday_weekly=Sun	# +%a
 backupday_monthly=01	# +%d
 
+timestamp=$(date "+%s")
 datefmt='%Y-%m-%d-%Hh%M'
-datetime=$(date "+${datefmt}")
+datetime=$(date -d @${timestamp} "+${datefmt}")
 
 # crontab(5) may define a restricted PATH
 PATH=/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin:/usr/local/sbin
@@ -144,14 +145,14 @@
 fi
 
 if [ "${archivemax_weekly}" -gt 0 ]; then
-	dayofweek=$(date -j -f ${datefmt} ${datetime} "+%a")
+	dayofweek=$(date -d @${timestamp} "+%a")
 	if [ "${dayofweek}" = "${backupday_weekly}" ]; then
 		backup weekly "${targets}"
 	fi
 fi
 
 if [ "${archivemax_monthly}" -gt 0 ]; then
-	dayofmonth=$(date -j -f ${datefmt} ${datetime} "+%d")
+	dayofmonth=$(date -d @${timestamp} "+%d")
 	if [ "${dayofmonth}" = "${backupday_monthly}" ]; then
 		backup monthly "${targets}"
 	fi
