--- tarsnaps.sh
+++ tarsnaps.sh
@@ -28,10 +28,11 @@
 backupday_weekly=Sun # +%a
 backupday_monthly=01 # +%d
 
+timestamp=$(date "+%s")
 datefmt=%Y-%m-%d-%Hh%M
-datetime=$(date "+${datefmt}")
-dayofweek=$(date -j -f "${datefmt}" "${datetime}" "+%a")
-dayofmonth=$(date -j -f "${datefmt}" "${datetime}" "+%d")
+datetime=$(date -d "@${timestamp}" "+${datefmt}")
+dayofweek=$(date -d "@${timestamp}" "+%a")
+dayofmonth=$(date -d "@${timestamp}" "+%d")
 
 die()
 {
